tools/integration_tests/run_e2e_tests.sh

```diff
@@ -42,10 +47,19 @@ fi
 if [ "$SKIP_NON_ESSENTIAL_TESTS_ON_PACKAGE" == true ]; then
   GO_TEST_SHORT_FLAG="-short"
   echo "Setting the flag to skip few un-important integration tests."
-  INTEGRATION_TEST_TIMEOUT=40m
+  INTEGRATION_TEST_TIMEOUT=50m
   echo "Changing the integration test timeout to: $INTEGRATION_TEST_TIMEOUT"
 fi
 
+if [ "$RUN_TESTS_WITH_PRESUBMIT_FLAG" == true ]; then
+  PRESUBMIT_RUN_FLAG="-presubmit"
+  if [ "$SKIP_NON_ESSENTIAL_TESTS_ON_PACKAGE" == true ]; then
```

Comment:
I believe these sub-checks are unnecessary. They won't affect the overall test execution time, but they do introduce redundant checks and logic. We can count above timeout for both the cases.